core/src/test/java/org/infinispan/api/APINonTxTest.java

```diff
@@ -632,4 +631,33 @@ public void testReplaceAll() {
       assertEquals("hello_es:hola", cache.get("es"));
       assertEquals("hello_cz:ahoj", cache.get("cz"));
    }
+
+   public void testFalseEqualsKeyGet() {
+      assertNull(cache.get(new FalseEqualsKey("boo", 1)));
```

Comment:
Hrmm maybe we should have this before and after the put? Since it will never be able to return it.